tests/utils/ringbuffer.py

```diff
@@ -0,0 +1,162 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""
+Tests for the `Ringbuffer` class
+"""
+
+import random
+from datetime import datetime, timedelta
+
+import numpy as np
+
+from frequenz.sdk.util.ringbuffer import (
+    OrderedRingBuffer,
+    RingBuffer,
+)
+
+
+def _test_simple_push_pop(container):
+    buffer = RingBuffer(container)
+
+    for i in range(len(container)):
+        buffer.push(i)
+
+    assert len(buffer) == len(container)
+
+    for i in range(len(container)):
+        assert i == buffer.pop()
+
+
+def test_simple_push_pop_list() -> None:
+    _test_simple_push_pop([0] * 50000)
+
+
+def test_simple_push_pop_array() -> None:
+    _test_simple_push_pop(np.empty(shape=(50000,), dtype=np.float64))
```

Comment:
`@pytest.parametrize()` is your friend: https://docs.pytest.org/en/6.2.x/parametrize.html

(untested but something like this should work)

```suggestion
@pytest.parametrize("container", [
    [0] * 50000,
    np.empty(shape=(50000,), dtype=np.float64),
])
def test_simple_push_pop(container):
    buffer = RingBuffer(container)

    for i in range(len(container)):
        buffer.push(i)

    assert len(buffer) == len(container)

    for i in range(len(container)):
        assert i == buffer.pop()
```